tools/gmhazard_scripts/db_creation/empirical_db/new_calc_emp_ds.py

```diff
@@ -101,95 +90,112 @@ def calculate_ds(
                 )
                 GMMs = empirical_factory.determine_all_gmm(
                     classdef.Fault(tect_type=classdef.TectType[tect_type]),
                     str(im),
                     model_dict,
                 )
                 for GMM_idx, (GMM, _) in enumerate(GMMs):
-                    db_type = f"{GMM.name}_{tect_type}"
-                    if GMM.name in ("K_20", "K_20_NZ", "ZA_06"):
+                    # TODO: Check those models
+                    if GMM.name in ("K_20", "K_20_NZ", "ZA_06", "CB_14", "ASK_14"):
                         continue
-                    print(f"Processing DB Type: {db_type}, {GMM_idx + 1} / {len(GMMs)}")
-
-                    for site in sites:
-                        rupture_context_df = get_rupture_context_df(
-                            distance_store,
-                            fault_df,
-                            site,
-                            site_df.loc[site],
-                            nhm_data,
-                            tect_type,
-                        )
-                        gmm_calculated_df = openquake_wrapper_vectorized.oq_run(
-                            GMM,
-                            classdef.TectType[tect_type],
-                            rupture_context_df,
-                            str(im),
-                            psa_periods if im is gc.im.IMType.pSA else None,
-                        )
-                        # Matching the index with rupture_df
-                        # to have a right rupture label
-                        gmm_calculated_df.set_index(
-                            rupture_df[
-                                rupture_df["rupture_name"].isin(
-                                    rupture_context_df["rupture_name"]
-                                )
-                            ].index,
-                            inplace=True,
-                        )
-                        gmm_calculated_df.columns = [
-                            f"{'_'.join(col.split('_')[:2]) if im is gc.im.IMType.pSA else im}_sigma"
-                            if col.endswith("std_Total")
-                            else col
-                            for col in gmm_calculated_df
-                        ]
-
-                        # Write an im_df to the given station/site
-                        imdb_dict[db_type].open()
-                        imdb_dict[db_type].add_im_data(
-                            site,
-                            gmm_calculated_df.loc[
-                                :,
-                                gmm_calculated_df.columns.str.endswith(
-                                    ("mean", "sigma")
+
+                    db_type = f"{GMM.name}_{tect_type}"
+                    suffix = f"_{suffix}" if suffix else ""
+                    # Create a DB if not exists
+                    imdb = gc.dbs.IMDBParametric(
+                        str(Path(output_dir) / f"{db_type}_ds{suffix}.db"),
```

Comment:
I think there's a function in utils called create_db_name?

Reply:
https://github.com/ucgmsim/gmhazard/blob/d99b40b261d2c6f44c64b8859b3b9ee99d8ae42b/calculation/gmhazard_calc/gmhazard_calc/utils.py#L409